cura/Settings/MachineManager.py

```diff
@@ -220,15 +220,16 @@ def _onGlobalPropertyChanged(self, key, property_name):
                                 # Remove from the value from the other stacks as well, unless the
                                 # top value from the other stacklevels is different than the new value
                                 for container in extruder_stack.getContainers():
-                                    if container == extruder_stack.getTop():
-                                        continue
+                                    # Commented out this line for a bit. Don't know why it's here, but it is causing
+                                    # issues with reset button. 
```

Comment:
Those two lines broke the reset button being handled properly, but removing it also created a recursion issue. I've since then fixed it.